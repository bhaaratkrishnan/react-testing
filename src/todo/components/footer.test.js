import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Footer } from "./footer";
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from "react-router-dom";

describe("Footer", () => {
  const todos = [
    { text: "Todo 1", completed: false },
    { text: "Todo 2", completed: false },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer todos={todos} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct number of active todos", () => {
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer todos={todos} />
      </MemoryRouter>
    );
    const todoCountElement = getByTestId("todo-count");
    expect(todoCountElement.textContent).toBe("2 items left!");
  });

  it('calls removeCompleted function when "Clear completed" button is clicked', () => {
    const removeCompletedMock = jest.fn();
    const todos =  [
      { text: "Todo 1", completed: true },
      { text: "Todo 2", completed: true },
      { text: "Todo 3", completed: false}
    ];
    const { getByText } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer todos={todos} dispatch={removeCompletedMock} />
      </MemoryRouter>
    );
    const clearCompletedButton = getByText("Clear completed");
    fireEvent.click(clearCompletedButton);
    expect(removeCompletedMock).toHaveBeenCalled();
  });

  it("completed tasks is disabled when there are no completed tasks", () => {
    const todos =  [
      { text: "Todo 1", completed: false },
      { text: "Todo 2", completed: false },
    ];
    const { getByText } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Footer todos={todos} />
      </MemoryRouter>
    );
    const clearCompletedButton = getByText("Clear completed");
    expect(clearCompletedButton).toBeDisabled();
  });
});
