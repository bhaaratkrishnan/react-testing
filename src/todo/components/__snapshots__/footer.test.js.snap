// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="footer"
    data-testid="footer"
  >
    <span
      class="todo-count"
      data-testid="todo-count"
    >
      2 items left!
    </span>
    <ul
      class="filters"
      data-testid="footer-navigation"
    >
      <li>
        <a
          class="selected"
          href="#/"
        >
          All
        </a>
      </li>
      <li>
        <a
          class=""
          href="#/active"
        >
          Active
        </a>
      </li>
      <li>
        <a
          class=""
          href="#/completed"
        >
          Completed
        </a>
      </li>
    </ul>
    <button
      class="clear-completed"
      disabled=""
    >
      Clear completed
    </button>
  </footer>
</DocumentFragment>
`;
